### ISSUE TITLE
Bug: H2 and H3 headings are not visually distinct

### ISSUE THREAD

**Issue by wbamberg** [mdn/yari#5485]
In the redesign, H2 and H3 headings look really similar. In the reference docs we rely heavily on H2 headings to communicate top-level structure (Description/Syntax/Specifications/Browser compatibility) to readers, and having it non-obvious whether something is an H2 or an H3 makes this difficult.

<img width="549" alt="Screen Shot 2022-03-04 at 4 23 19 PM" src="https://user-images.githubusercontent.com/432915/156859021-9a8433df-cd18-4d45-a178-d6972e1b1680.png">

**Comment by e3dio** [comment]
This is a common complaint, duplicate of https://github.com/mdn/yari/issues/5466

The old design was popular because it did a great job making the different sections stand out:

![image](https://user-images.githubusercontent.com/85405955/156891307-d7bcdd99-0eea-4367-8d73-7d824a885662.png)

The new design takes the opposite approach with header boldness, the smaller headers get more bold instead of less bold at each level and they look similar, impossible to tell them apart:

![image](https://user-images.githubusercontent.com/85405955/156891355-e4469527-ecc7-4296-9b3f-257aa5dd4b89.png)

My example fix at https://github.com/mdn/yari/issues/5389 uses boldness and indentation to make them stand out, while being more vertically compact to display more information:

![image](https://user-images.githubusercontent.com/85405955/156891736-400a5cb3-4bbb-4e18-bde1-79314db20a4a.png)

Also note the new design is harder to see the code blocks with less contrast, and is missing the border-left color

**Comment by schalkneethling** [comment]
@wbamberg What do you think of the below change?

![Screenshot of new typography hierarchy](https://user-images.githubusercontent.com/10350960/159332098-7da75eec-191c-41b3-b97a-2bdfe7b466ce.png)

**Comment by schalkneethling** [comment]
> @schalkneethling I think this looks great! Do you have a branch where I could try it out?

I am spinning up a dev instance. Will let you know as soon as it is live(so to speak).

**Comment by schalkneethling** [comment]
Hey @wbamberg, we are trialing some improvements to the readability and accessibility of MDN Web Docs and we want your input. Please join in the discussion here: https://github.com/mdn/yari/discussions/5715
